providers/base/units/graphics/jobs.pxu

```diff
@@ -544,6 +544,7 @@ id: graphics/nvlink-status-check
 requires:
  dmi.product in ['Desktop','Low Profile Desktop','Tower','Mini Tower', 'Space-saving']
  graphics_card.driver == 'nvidia'
+ graphics_card.gpu_count >= "2"
```

Comment:
I don't think it will work as expected. The `gpu_count` provided by this resource job includes both discrete and integrated GPU, like on this platform: https://ubuntu.com/certified/202212-30934

One Nvidia card was present bu the resource reported two GPU:

```
bus: pci
category: VIDEO
driver: i915
gpu_count: 2
index: 1
path: /devices/pci0000:00/0000:00:02.0
prime_gpu_offload: Off
product: PCI ID 0xa7a0
product_id: 42912
product_slug: PCI_ID_0xa7a0
subproduct_id: 3074
subvendor_id: 4136
switch_to_cmd: prime-select intel
vendor: Intel Corporation
vendor_id: 32902
vendor_slug: Intel_Corporation

bus: pci
category: VIDEO
driver: nvidia
gpu_count: 2
index: 2
path: /devices/pci0000:00/0000:00:06.2/0000:01:00.0
prime_gpu_offload: Off
product: PCI ID 0x25bb
product_id: 9659
product_slug: PCI_ID_0x25bb
subproduct_id: 3074
subvendor_id: 4136
switch_to_cmd: prime-select nvidia
vendor: NVIDIA Corporation
vendor_id: 4318
vendor_slug: NVIDIA_Corporation
```

Reply:
@yphus 
You are right. This resource job count all gpu on the DUT, including iGPU and dGPU.
What do you think if I modify this jobs to let it output the count od iGPU and dGPU?

Reply:
@LiaoU3 it would be extremely helpful indeed. If you know how to do it please go ahead and add new key/value property to the [resource job](https://github.com/canonical/checkbox/blob/main/providers/resource/bin/graphics_card_resource.py) output.

Reply:
This would be very nice, but it's gonna be tricky I think.

- On a desktop with an AMD graphics card, is this AMD GPU an "integrated" GPU or a "discrete" one?
- You could think that Intel GPUs are always integrated, but there are now Intel GPUs such as [Intel Arc A750](https://www.phoronix.com/review/intel-a750-a770-arc-linux) that are separate graphics cards...

Also, the way Checkbox handles resource jobs might not be what you expect.

In the sample above, the device has one Intel GPU and one Nvidia GPU. `graphics_card.driver == 'nvidia'` does **not** mean "run this job only for the graphics card that uses the nvidia driver", it means "run this job if **any** of the grpahics cards present on this device uses the nvidia driver". It's a subtle difference, but it makes a big difference when you're writing jobs...

Reply:
This job is to test the nvlink, so maybe I could have the resource job count all the connected nvidia card. And also it's easy to fulfill,
```
ubuntu@ubuntu:~/oem-qa-checkbox-installer$ nvidia-smi -L
GPU 0: NVIDIA RTX A4500 (UUID: GPU-5b1b2f62-281c-fcda-59a8-65885b5df254)
GPU 1: NVIDIA RTX A4500 (UUID: GPU-405c9741-e9ea-1fd8-ff03-d01cf4d71fc8)
```
```
ubuntu@ubuntu:~/oem-qa-checkbox-installer$ nvidia-smi -L | wc -l
2
```

Reply:
is `nvidia-smi` always available if you install the nv drivers or is it a separate tool package? If it's there by default writing a dedicated resource job for graphics/nvlink-status-check running the above command could do it.

Reply:
the nvidia-smi is part of the nvidia-utils package, and it would be installed by default while installing the nvidia-driver package.
@yphus Thanks for the suggestion. @LiaoU3  let's do it this way.

Reply:
Now I use the command `lspci | grep NVIDIA | grep VGA -c` instead of `nvidia-smi -L | wc -l` to prevent from SW accidentally forget attatching the nvidia utility. Using the first command can solve this kind of problem and also get the correct NVIDIA GPU count.